src/libinjector/linux/methods/linux_write_file.c

```diff
@@ -298,13 +301,22 @@ event_response_t handle_write_file(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
 
 }
 
-static size_t read_chunk(injector_t injector, FILE* fp)
+static bool read_chunk(injector_t injector, size_t* bytes_read)
 {
-    injector->buffer.len = fread(injector->buffer.data, 1, FILE_BUF_SIZE, fp);
+    if (!bytes_read)
```

Comment:
can't this be replaced by `injector->buffer.len` only?

Reply:
plz tell what do you mean?

Reply:
smth like this before returning?

    if (bytes_read)
        *bytes_read = injector->buffer.len;

Reply:
or I can actually make the bytes_read parameter dissapear, that's good idea too

Reply:
static bool read_chunk(injector_t injector)
{
    injector->buffer.len = fread(injector->buffer.data, 1, FILE_BUF_SIZE, injector->fp);
    if (ferror(injector->fp))
    {
        PRINT_DEBUG("Failed to read from file\n");

        injector->buffer.len = 0;
        return false;
    }
    injector->buffer.total_processed += injector->buffer.len;
    PRINT_DEBUG("Chunk read successful (%ld/%ld)\n", injector->buffer.total_processed, injector->buffer.total_len);
    
    return true;
}
like this

Reply:
I think at the end `bytes_read` was being assigned equal to `injector->buffer.len` and I couldn't see any other code that'll change `bytes_read` in other places. So they are essentially the same, though now that I think about it. Passing an argument to keep track of it seems better. Sorry to bother

Reply:
> or I can actually make the bytes_read parameter dissapear, that's good idea too

I'll leave this decision up to you now, both seem fine to me